networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/config/GossipScoringConfigurator.java

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.networking.eth2.gossip.config;
+
+import java.time.Duration;
+import java.util.Optional;
+import tech.pegasys.teku.networking.eth2.gossip.AggregateGossipManager;
+import tech.pegasys.teku.networking.eth2.gossip.AttesterSlashingGossipManager;
+import tech.pegasys.teku.networking.eth2.gossip.BlockGossipManager;
+import tech.pegasys.teku.networking.eth2.gossip.ProposerSlashingGossipManager;
+import tech.pegasys.teku.networking.eth2.gossip.VoluntaryExitGossipManager;
+import tech.pegasys.teku.networking.eth2.gossip.topics.TopicNames;
+import tech.pegasys.teku.networking.p2p.gossip.config.GossipConfig;
+import tech.pegasys.teku.networking.p2p.gossip.config.GossipScoringConfig;
+import tech.pegasys.teku.networking.p2p.gossip.config.GossipTopicScoringConfig;
+import tech.pegasys.teku.networking.p2p.gossip.config.GossipTopicsScoringConfig;
+import tech.pegasys.teku.spec.constants.SpecConstants;
+import tech.pegasys.teku.ssz.SSZTypes.Bytes4;
+
+/**
+ * These calculations have been derived following the gossip scoring implementation from Lighthouse
+ * (which is Apache 2.0 licensed). Big thanks to the Lighthouse team!
+ *
+ * @see <a
+ *     href="https://github.com/sigp/lighthouse/blob/f183af20e3fb44773397c9e26540703ee94ec366/beacon_node/eth2_libp2p/src/behaviour/gossipsub_scoring_parameters.rs">Sigma
+ *     Prime's Lighthouse gossip scoring logic</a>
+ */
+class GossipScoringConfigurator implements GossipConfigurator {
+  private static final int GOSSIP_D = 8;
+  private final ScoringConfig scoringConfig;
+
+  public GossipScoringConfigurator(final SpecConstants specConstants) {
+    this.scoringConfig = ScoringConfig.create(specConstants, GOSSIP_D);
+  }
+
+  @Override
+  public void configure(
+      final GossipConfig.Builder gossipConfigBuilder, final Eth2Context eth2Context) {
+    gossipConfigBuilder
+        .d(GOSSIP_D)
+        .dLazy(GOSSIP_D)
+        .dLow(6)
```

Comment:
Do I understand correctly that other gossip options are picked up from the `GossipConfig` in `p2p` module? 
If yes, may be it worth removing them from the `GossipConfig` (as the `p2p` module assumed to be Eth2-agnostic) and put them here explicitly?
I think there is no need to set up any Gossip defaults in `p2p` as libp2p implicitly setting defaults for gossip

Reply:
Moved these constants up to this class.

Reply:
Actually, ended up undoing this change - we should either have set defaults in `GossipConfig` (so we don't have to manually configure these everywhere) or else make all of the params `Optional` so we can use the jvm-libp2p defaults.   Simplest thing seems to be to leave this as-is for now.